sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs

```diff
@@ -150,5 +152,211 @@ Page<TimeSeriesHierarchy> NextPageFunc(string nextLink, int? pageSizeHint)
                 throw;
             }
         }
+
+        /// <summary>
+        /// Gets Time Series Insights hierarchies by hierarchy names asynchronously.
+        /// </summary>
+        /// <param name="timeSeriesHierarchyNames">List of names of the Time Series hierarchies to return.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>
+        /// List of hierarchy or error objects corresponding by position to the array in the request.
+        /// Hierarchy object is set when operation is successful and error object is set when operation is unsuccessful.
+        /// </returns>
+        /// <remarks>
+        /// For more samples, see <see href="https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/samples">our repo samples</see>.
+        /// </remarks>
+        /// <exception cref="ArgumentNullException">
+        /// The exception is thrown when <paramref name="timeSeriesHierarchyNames"/> is <c>null</c>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// The exception is thrown when <paramref name="timeSeriesHierarchyNames"/> is empty.
+        /// </exception>
+        public virtual async Task<Response<TimeSeriesHierarchyOperationResult[]>> GetByNameAsync(
+            IEnumerable<string> timeSeriesHierarchyNames,
```

Comment:
So the input is an Enumerable which is not guaranteed to maintain the order of the items. The comments mention that the order of the result array corresponds to that of the request, how are you enforcing that?

Reply:
As per our discussion with Azure SDK team

> "You should still use IEnumerable<T> for parameters because it lets customers pass an array, a list, a readonly list returned from another operation, something they pulled out of a LINQ query, etc.  IEnumerable<T> isn’t as easily indexed, but it’s still ordered"